argus/cbinit/base_config_generator.py

```diff
@@ -0,0 +1,115 @@
+#    under the License.
+import abc
+import ConfigParser
+import ntpath
+import StringIO
+import tempfile
+
+from argus import util
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseCBInitConfig(object):
+    """Class that abstract the CLoudBase-Init config files.
+
+    For any config type we need to know :
+    :param default_config:
+        Name of the default config from argus/resources
+    :param name:
+        Name of the file in the instance
+    """
+
+    default_config = None
+    config_name = None
+
+    SERVICES = {
+        util.HTTP_SERVICE:         "httpservice.HttpService",
+        util.CONFIG_DRIVE_SERVICE: "configdrive.ConfigDriveService",
+        util.EC2_SERVICE:          "ec2service.EC2Service",
+        util.OPEN_NEBULA_SERVICE:  "opennebulaservice.OpenNebulaService",
+        util.CLOUD_STACK_SERVICE:  "cloudstack.CloudStack",
+        util.MASS_SERVICE:         "maasservice.MaaSHttpService"
+    }
+
+    """An object that holds the cloudbaseinit config."""
+
+    def __init__(self, client):
+        """For the CBInitConfig we need a client connected to the instance.
+
+        :param client:
+            A client connected to the instance.
+        """
+        self._client = client
+        self.conf = self._get_base_conf(self.default_config)
+
+    @staticmethod
+    def _get_base_conf(config_name):
+        """Return a ConfigParser object withe default values."""
+        base_conf = StringIO.StringIO(
+            util.get_resource(config_name))
+        conf = ConfigParser.ConfigParser()
+        conf.readfp(base_conf)
+        return conf
+
+    @abc.abstractmethod
+    def set_conf_value(self, name, value):
+        """Set a config value in Default section."""
+        pass
+
+    @abc.abstractmethod
+    def config_specific_paths(self):
+        """Populate the ConfigParser object with instance specific values."""
+        pass
+
+    def _get_service(self, service_type):
+        """Returns the cloudbase-init config value.
+
+        :param service_type:
+            This can be http, configfrive, ec2, opennebula,
+            cloudstacl or mass.
+        """
+        return '.'.join([util.SERVICES_PREFIX, self.SERVICES[service_type]])
+
+    def set_service_type(self, service_type):
+        """Set the service type config.
+
+        :param service_type:
+            This can be a string like http, configfrive, ec2, opennebula,
+            cloudstacl, mass or it can be a list withe the required service
+            types. The order matters!
+
+        ::
+        Example:
+        ::
+            [uti.MASS, util.EC2_SERVICE ...]
+        """
+        if isinstance(service_type, str):
+            service_type = [service_type]
+
+        service_type = [self._get_service(serv) for serv in service_type]
+        conf_value = ",".join(service_type)
+        self.set_conf_value("metadata_services", conf_value)
+
+    def apply_config(self, path):
+        """Set the LocalScript path."""
```

Comment:
Rephrase the docstring. Also add more information related to the `path` argument.